client/lib/mixins/close-on-esc/hoc.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import { filter, last } from 'lodash';
+import React, { PureComponent } from 'react';
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'esc-hoc' );
+let components = [];
+
+function startCloseOnEscForComponent( component ) {
+	components.push( component );
+
+	debug( 'start', components.length );
+
+	if ( components.length > 0 ) {
```

Comment:
Agreed, I didn't want to tinker with the original behaviour of the mixin too much at this stage though!
That said, `addKeydownListener` calls the following:
```es6
document.addEventListener( 'keydown', onKeydown, true );
```
Which I believe will not create duplicate listeners since the arguments are always the same?
I checked that `onKeydown` was not called multiple times after `addKeydownListener` had been called repeatedly and it was only called once per keydown 👍 (Just checked by adding a log)